assets/model_monitoring/components/src/genai_token_statistics_compute_metrics/metrics_processor.py

```diff
@@ -54,127 +38,103 @@ def create_span_tree_from_dataframe_row(self, row: Row) -> Optional[SpanTreeNode
     def create_span_tree_from_dataframe(self, df: DataFrame) -> List[Optional[SpanTreeNode]]:
         """Create a list of SpanTrees from a DataFrame."""
         # Convert the DataFrame to an RDD and then to a list of Rows
-        rows = df.rdd.map(lambda row: row).collect()
-
-        # Create a SpanTree for each row
-        span_trees = [self.create_span_tree_from_dataframe_row(row) for row in rows]
-        return span_trees
+        df.rdd.foreach(self.process_row)
 
     def process(self, df_traces: DataFrame):
         """Process the aggregated trace data to compute metrics.
 
         Args:
             df_traces (DataFrame): Aggregated traces from the GenAI processor.
         """
+        print("process")
```

Comment:
Can we have more clear, user friendly information/message?

Reply:
I'll remove it